### PR TITLE
Feat/scrollbar and datazoom

### LINKED ISSUE #965: [Feature] scrollbar enhance
### What problem does this feature solve?

enhance about scrollbar in LAS：
https://bytedance.feishu.cn/docx/KfZndDQBSoJMHrxq6U3cJPZsnjg

- drag：rate、reverse
- zoom: rate、focus
- scroll: rate、reverse

### What does the proposed API look like?
enableInteraction()
disableInteraction()
zoomIn()
zoomOut()

[null](https://bytedance.feishu.cn/docx/KfZndDQBSoJMHrxq6U3cJPZsnjg)

### LINKED ISSUE #1082: [Feature] datazoom enhance
### What problem does this feature solve?

align with echarts：
image

### What does the proposed API look like?

null

### PR DESCRIPTION
**Pull request by skie1997**
[[中文版模板 / Chinese template](https://github.com/VisActor/VChart/blob/main/.github/PULL_REQUEST_TEMPLATE/pr_cn.md?plain=1)]

### 🤔 This is a ...

- [ ] New feature
- [ ] Bug fix
- [ ] TypeScript definition update
- [ ] Bundle size optimization
- [ ] Performance optimization
- [x] Enhancement feature
- [ ] Refactoring
- [ ] Update dependency
- [ ] Code style optimization
- [ ] Test Case
- [ ] Branch merge
- [ ] Release
- [ ] Site / documentation update
- [ ] Demo update
- [ ] Workflow
- [ ] Other (about what?)

### 🔗 Related PR link
close #1082 
close #965 

### 🐞 Bugserver case id
6511e2af172872291c4615d7
6511e268172872291c4615d6
6511e1b4172872291c4615d4
6511e1ee172872291c4615d5

### 💡 Background and solution
20230926225103_rec_
20230926225216_rec_
20230926224949_rec_
20230926225030_rec_

### 📝 Changelog

| Language   | Changelog |
| ---------- | --------- |
| 🇺🇸 English |           |
| 🇨🇳 Chinese |           |

### ☑️ Self-Check before Merge

⚠️ Please check all items below before requesting a reviewing. ⚠️

- [ ] Doc is updated/provided or not needed
- [ ] Demo is updated/provided or not needed
- [ ] TypeScript definition is updated/provided or not needed
- [ ] Changelog is provided or not needed

---

### 🚀 Summary

copilot:summary

### 🔍 Walkthrough

copilot:walkthrough

### CODE REVIEW

File: packages/vchart/src/component/data-zoom/data-filter-base-component.ts [hunk 1]
```diff
@@ -383,9 +440,29 @@ export abstract class DataFilterBaseComponent<T extends IDataFilterComponentSpec
   setAttrFromSpec() {
     super.setAttrFromSpec();
 
-    if (this._spec.roam) {
+    // interaction相关
+    if (isValid(this._spec.roamZoom) && this._spec.roamZoom) {
```
Comment (kkxxkk2019) [inline packages/vchart/src/component/data-zoom/data-filter-base-component.ts:444]: 不需要判断 isValid 吧

File: packages/vchart/src/component/data-zoom/data-filter-base-component.ts [hunk 2]
```diff
@@ -383,9 +440,29 @@ export abstract class DataFilterBaseComponent<T extends IDataFilterComponentSpec
...
+      this._zoomAttr = merge({}, this._zoomAttr, this._spec.roamZoom);
+    } else {
+      this._zoomAttr.enable = false;
+    }
+
+    if (isValid(this._spec.roamDrag) && this._spec.roamDrag) {
```
Comment (kkxxkk2019) [inline packages/vchart/src/component/data-zoom/data-filter-base-component.ts:450]: 这里也是

File: packages/vchart/src/component/data-zoom/data-filter-base-component.ts [hunk 3]
```diff
@@ -383,9 +440,29 @@ export abstract class DataFilterBaseComponent<T extends IDataFilterComponentSpec
...
+      this._dragAttr = merge({}, this._dragAttr, this._spec.roamDrag);
+    } else {
+      this._dragAttr.enable = false;
+    }
+
+    if (isValid(this._spec.roamScroll) && this._spec.roamScroll) {
```
Comment (kkxxkk2019) [inline packages/vchart/src/component/data-zoom/data-filter-base-component.ts:456]: same

File: packages/vchart/src/component/data-zoom/data-filter-base-component.ts [hunk 4]
```diff
@@ -448,6 +525,16 @@ export abstract class DataFilterBaseComponent<T extends IDataFilterComponentSpec
     this._endValue = this._statePointToData(end);
     this._start = start;
     this._end = end;
+    this._minSpan = this._spec.minSpan ?? 0;
+    this._maxSpan = this._spec.maxSpan ?? 1;
+    if (this._spec.minValueSpan && isContinuous(this._stateScale.type)) {
+      this._minSpan = this._spec.minValueSpan / (this._stateScale.domain()[1] - this._stateScale.domain()[0]);
```
Comment (kkxxkk2019) [inline packages/vchart/src/component/data-zoom/data-filter-base-component.ts:531]: 分母为 0 的场景处理下

File: packages/vchart/src/component/data-zoom/interface.ts [hunk 5]
```diff
@@ -76,17 +77,118 @@ export interface IDataFilterComponentSpec extends Omit<IComponentSpec, 'width' |
...
-  roam?: boolean;
   /**
    * 是否为自动模式。开启以后，组件不会导致轴 scale 缩放，end、roam 等可能导致缩放的配置将被忽略，且组件可以自动消失
    * @since 1.4.0
    */
   auto?: boolean;
+  /**
+   * 是否锁定选择区域（或叫做数据窗口）的大小
+   * @default false
+   * @since 1.5.0
```
Comment (kkxxkk2019) [inline packages/vchart/src/component/data-zoom/interface.ts:88]: 1.5.1

File: packages/vchart/src/component/data-zoom/interface.ts [hunk 6]
```diff
@@ -76,17 +77,118 @@ export interface IDataFilterComponentSpec extends Omit<IComponentSpec, 'width' |
    * 是否进行自动缩进
    */
   autoIndent?: boolean;
-  /**
-   * 是否开启鼠标缩放和平移漫游。默认不开启
-   */
-  roam?: boolean;
```
Comment (kkxxkk2019) [inline packages/vchart/src/component/data-zoom/interface.ts:82]: 这个属性不支持了的话属于不兼容修改了，这个对应新的属性是啥？要不要做下兼容

File: packages/vchart/src/component/data-zoom/data-filter-base-component.ts [hunk 7]
```diff
@@ -99,14 +127,42 @@ export abstract class DataFilterBaseComponent<T extends IDataFilterComponentSpec
...
+  enableInteraction() {
+    this._activeRoam = true;
+  }
+  disableInteraction() {
+    this._activeRoam = false;
+  }
+  zoomIn(location?: { x: number; y: number }) {
+    this._handleChartZoom({
+      zoomDelta: 1.2, // 经验值
+      zoomX: location?.x,
```
[the remark this answers is not in the document]
Comment (skie1997) [reply packages/vchart/src/component/data-zoom/data-filter-base-component.ts:139]: 在后面的逻辑里会处理不存在的情况，如果没有传的话，focLocation为画布中心
image

File: packages/vchart/src/component/data-zoom/data-filter-base-component.ts (hunk 1 above)
Comment (skie1997) [reply packages/vchart/src/component/data-zoom/data-filter-base-component.ts:444]: 如果用户没有配置this._spec.roamZoom，enable默认为false；如果配置this._spec.roamZoom的话，enable默认为true。两种情况的默认值不一样，所以判断了一下

File: packages/vchart/src/component/data-zoom/interface.ts (hunk 6 above)
Comment (skie1997) [reply packages/vchart/src/component/data-zoom/interface.ts:82]: 对应的新属性是roamDrag、roamScroll、roamZoom，需要做下兼容的，我处理下

File: packages/vchart/src/component/data-zoom/data-filter-base-component.ts (hunk 4 above)
Comment (skie1997) [reply packages/vchart/src/component/data-zoom/data-filter-base-component.ts:531]: get～

**Review by kkxxkk2019 — APPROVED** [review]

**Pull request merged**